searx/favicons/resolvers.py

```diff
@@ -0,0 +1,81 @@
+# SPDX-License-Identifier: AGPL-3.0-or-later
+"""Implementations of the favicon *resolvers* that are available in the favicon
+proxy."""
+
+from __future__ import annotations
+
+__all__ = ['RESOLVERS', 'RESOLVER_MAP']
+
+from typing import Callable
+from searx import network
+
+RESOLVERS: list[str]
+RESOLVER_MAP: dict[str, Callable]
+
+
+def allesedv(domain, req_args):
+    """Favicon Resolver from allesedv.com / https://favicon.allesedv.com/"""
+    data, mime = (None, None)
+
+    # will just return a 200 regardless of the favicon existing or not
+    # sometimes will be correct size, sometimes not
+    response = network.get(f"https://f1.allesedv.com/32/{domain}", **req_args)
+    if response and response.status_code == 200:
+        mime = response.headers['Content-Type']
+        if mime != 'image/gif':
+            data = response.content
+    return data, mime
+
+
+def duckduckgo(domain, req_args):
+    """Favicon Resolver from duckduckgo.com / https://blog.jim-nielsen.com/2021/displaying-favicons-for-any-domain/"""
+    data, mime = (None, None)
+
+    # will return a 404 if the favicon does not exist and a 200 if it does,
+    response = network.get(f"https://icons.duckduckgo.com/ip2/{domain}.ico", **req_args)
+    if response and response.status_code == 200:
+        # api will respond with a 32x32 png image
+        mime = response.headers['Content-Type']
+        data = response.content
+    return data, mime
+
+
+def google(domain, req_args):
+    """Favicon Resolver from google.com"""
+    data, mime = (None, None)
+
+    # URL https://www.google.com/s2/favicons?sz=32&domain={domain}" will be
+    # redirected (HTTP 301 Moved Permanently) to t1.gstatic.com/faviconV2:
+    response = network.get(
+        f"https://t1.gstatic.com/faviconV2?client=SOCIAL&type=FAVICON&url=https://{domain}&size=32",
```

Comment:
Looks like this only works with some websites.

Doesn't work:
https://t1.gstatic.com/faviconV2?client=SOCIAL&type=FAVICON&url=https://reddit.com&size=32

For it to load needs the `fallback_opts` param:
https://t0.gstatic.com/faviconV2?client=SOCIAL&type=FAVICON&fallback_opts=TYPE,SIZE,URL&url=http://reddit.com&size=32


But some (actually most) sites work fine as:
https://t1.gstatic.com/faviconV2?client=SOCIAL&type=FAVICON&url=https://priv.au&size=32

and from my brief testing, they also do with the `fallback_opts` param, so maybe this should be included:

https://t1.gstatic.com/faviconV2?client=SOCIAL&type=FAVICON&fallback_opts=TYPE,SIZE,URL&url=https://priv.au&size=32

Reply:
> For it to load needs the `fallback_opts` param:

Yeah this is also what the redirect https://www.google.com/s2/favicons?sz=32&domain=reddit.com does .. thought I can shorten the URL a little bit .. but I can't :-)

Fixed and squashed into the WIP commit on top of the branch [diff](https://github.com/searxng/searxng/compare/e63e83b1d1688201f338b01a0187c5a3c162ef7f..f3acc085b709a9aba45ddb801a85466d56d1fbe4).